### PR TITLE
Fix Tmove

### PR DESCRIPTION
**Pull request by ye-luo**
1) A bug was introduced in PR #289 affecting Tmove in both AoS and SoA cases.
This fix is validated against v3.1.1.
The current tests labelled dmcnl are all short tests. Need long ones.

2) Add ENABLE_SOA to ctest_script.cmake

### CODE REVIEW

**Comment by markdewing** [comment]
okay to test

File: CMake/ctest_script.cmake [hunk 1]
```diff
@@ -74,6 +74,12 @@ ELSEIF( ${HOSTNAME} MATCHES "billmp1" )
     ELSE()
         SET( CTEST_BUILD_NAME Clang-Release-Mac )
     ENDIF()
+ELSEIF( ${HOSTNAME} MATCHES "ryzen-box" )
```
Comment (markdewing) [inline CMake/ctest_script.cmake]: Did you intend to check in this code?

Comment (ye-luo) [inline CMake/ctest_script.cmake]: Yes. Maintaining a local change is painful.

Comment (prckent) [inline CMake/ctest_script.cmake]: OK if it makes development+testing easier. It does raise the question about why these settings are needed, but that could be asked of any of the machine settings in the ctest script...

File: src/QMCDrivers/DMC/DMCUpdatePbyPFast.cpp [hunk 2]
```diff
@@ -192,22 +189,27 @@ void DMCUpdatePbyPWithRejectionFast::advanceWalker(Walker_t& thisWalker, bool re
...
-      Psi.acceptTMove(W,iat,w_buffer);
-#if 0
-      RealType ratio=Psi.ratio(W,iat,dG,dL);
-      W.G += dG;
-      W.L += dL;
-      //thisWalker.R[iat]=W.R[iat];
-      //w_buffer.rewind();
-      //W.copyToBuffer(w_buffer);
-      RealType logpsi = Psi.evaluateLog(W,w_buffer);
+#ifdef ENABLE_SOA
```
Comment (prckent) [inline src/QMCDrivers/DMC/DMCUpdatePbyPFast.cpp]: Please check the reptation implementation also. See @rcclay 's comment about earlier tests.

Comment (ye-luo) [inline src/QMCDrivers/DMC/DMCUpdatePbyPFast.cpp]: RMC doesn't have Tmove enabled. The UseTMove was disabled by @rcclay in 2014.
pbyp and allp moves of AoS and SoA builds are both checked against v3.1.1. See my comment on #229

Comment (prckent) [inline src/QMCDrivers/DMC/DMCUpdatePbyPFast.cpp]: Good. So Tmoves will need investigating by a future reptation developer/user, and they can't be run accidentally.

**Pull request merged**